design/layer2-bind-interfaces.md

```diff
@@ -0,0 +1,56 @@
+# Configuring announcing interfaces
+
+## Summary
+
+The purpose of this enhancement is to make Speakers only announce VIPs on specified network interfaces.
+
+Previous discussions about this topic can be found in [this issue](https://github.com/metallb/metallb/issues/277).
+
+## Motivation
+
+A system may have multiple networks based on functions, and thus people want MetalLB only works on specified networks.
+In BGP mode, this situation can be handled by BGP configuration, while in L2 mode,  there is no way to achieve this purpose.
+
+[<img src="./images/example_system.png" width="800"/>](./images/example_system.png)
+
+### Goals
+- The network interfaces the MetalLB uses to broadcast VIP can be specified.
+
+### Non-Goals
+- Change any core functions of MetalLB
+
+## Data model
+
+Based on the current MetalLB CRD design (`8d906f0abef3d7ed5b0435fa4edfbe5e37b38557`), we'd like to add a new field `BindInterfaces` to `L2Advertisement`.
+
+`BindInterfaces` describes network interface names of the Nodes MetalLB should use. It is a list of interfaces' names, and it is optional. Once users set "bindInterface", then Speakers should only create announcers on these interfaces. Otherwise, Speakers keep the default behavior which creates announcers on every interface.
+
+The new definition of `L2Advertisement` is:
+```go
+type L2Advertisement struct {
+  Name                     string `yaml:"name"`
+  IPAddressPools           []string `yaml:"ipAddressPools"`
+  IPAddressPoolSelector    *metav1.LabelSelector `yaml:"ipAddressPoolSelector"`
+  BindInterfaces           []string `yaml:"bindInterfaces"`
+}
+```
+
+An example definition can be:
+```yaml
+apiVersion: metallb.io/v1beta1
+kind: L2Advertisement
+metadata:
+  name: example-advertisement
+  namespace: metallb-system
+spec:
+    ipaddresspools:
+      - pool1
+      - pool2
+    bindInterfaces:
+      - eth0
+      - eth1
+```
+
+## Details
+
+When the property `bindInterfaces` of L2Advertisement changed, Speaker will close or open ARP and NDP responders according to the "bindInterface".
```

Comment:
What happens when we specify a selector that doesn't include any interfaces on a given node?
Do we have any plans?

Reply:
In this scenario, the Speaker will not open any responders. Maybe we can add a condition type of CRD status for it.

Reply:
Let's be explicit about the fact that the absence of a valid interface won't influence the way we elect the speaker for a given ip.

Reply:
I has explicit it in the "Non-Goals" part of the second commit